pkg/handler/on_demand.go

```diff
@@ -48,11 +48,6 @@ func (h *onDemandResourceHandler) HandleCreate(requestCount int, pod *v1.Pod) (c
 
 // HandleDelete reclaims the on demand resource by passing the Delete Job to the respective Worker
 func (h *onDemandResourceHandler) HandleDelete(pod *v1.Pod) (ctrl.Result, error) {
-	if _, ok := pod.Annotations[h.resourceName]; !ok {
```

Comment:
Yeah, this is done for cases when termination grace period is set to 0 and Pod is immediately created and deleted. In that case I observed the delete event didn't have the annotation. This will query the local DS to see if Pod had been allocated a resource.